### PR TITLE
add azure assistants example

### PR DESCRIPTION
**Pull request by glharper**
- [ x] I understand that this repository is auto-generated and my pull request may not be merged

## Changes being requested
Added code example for using Azure Assistants

### CODE REVIEW

File: examples/assistants-azure.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env -S npm run tsn -T
+
+import { AzureOpenAI } from 'openai';
+
+/**
+ * Example of polling for a complete response from an assistant
+ */
+const apiVersion = '2024-02-15-preview';
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:8]: Should we move to a recent API version?
```suggestion
const apiVersion = '2024-05-01-preview';
```

File: examples/assistants-azure.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const openai = new AzureOpenAI({
+  endpoint:'https://YOUR_AZURE_OPENAI_RESOURCE_NAME.openai.azure.com/',
+  apiKey: 'YOUR_AZURE_API_KEY',
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:11]: Should we use Entra ID auth instead?

File: examples/assistants-azure.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const run = await openai.beta.threads.runs.createAndPoll(thread.id, {
+    assistant_id: assistantId,
+    additional_instructions: 'Please address the user as Jane Doe. The user has a premium account.',
+  });
+
+  console.log('Run finished with status: ' + run.status);
+
+  if (run.status == 'completed') {
+    const messages = await openai.beta.threads.messages.list(thread.id);
+    for (const message of messages.getPaginatedItems()) {
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:48]: Should we use the `for await` pattern here?
```suggestion
    for await (const message of messages) {
```

File: examples/assistants-azure.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    messages: [
+      {
+        role: 'user',
+        content: '"I need to solve the equation `3x + 11 = 14`. Can you help me?"',
+      },
+    ],
+  });
+
+  let threadId = thread.id;
+  console.log('Created thread with Id: ' + threadId);
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:37]: nitpick: simplifying:
```suggestion
  console.log('Created thread with Id: ' + thread.id);
```

File: examples/assistants-azure.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+async function main() {
+  const assistant = await openai.beta.assistants.create({
+    model: 'gpt-4',
+    name: 'Math Tutor',
+    instructions: 'You are a personal math tutor. Write and run code to answer math questions.',
+    // tools = [],
+  });
+
+  let assistantId = assistant.id;
+  console.log('Created Assistant with Id: ' + assistantId);
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:25]: nitpick: simplifying:
```suggestion
  console.log('Created Assistant with Id: ' + assistant.id);
```

File: examples/assistants-azure.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        content: '"I need to solve the equation `3x + 11 = 14`. Can you help me?"',
+      },
+    ],
+  });
+
+  let threadId = thread.id;
+  console.log('Created thread with Id: ' + threadId);
+
+  const run = await openai.beta.threads.runs.createAndPoll(thread.id, {
+    assistant_id: assistantId,
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:40]:
```suggestion
    assistant_id: assistant.id,
```

File: examples/assistants-azure.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  apiVersion
+});
+
+
+async function main() {
+  const assistant = await openai.beta.assistants.create({
+    model: 'gpt-4',
+    name: 'Math Tutor',
+    instructions: 'You are a personal math tutor. Write and run code to answer math questions.',
+    // tools = [],
```
Comment (deyaaeldeen) [inline examples/assistants-azure.ts:21]: Do we need to use tools?
```suggestion
```

**Review by deyaaeldeen — COMMENTED** [review]
LGTM, I left mostly nitpicks.

**Review by deyaaeldeen — APPROVED** [review]

**Comment by rattrayalex** [comment]
Sorry this took so long to merge – thanks so much for the contribution!

**Pull request merged**